File: Orbit.Domain/DTOs/MealPlanner/GetRecipesResponse.cs
namespace Orbit.Domain.DTOs.MealPlanner
{
    public class GetRecipesResponse
    {
        public required RecipeItem[] Recipes { get; set; }
    }

    public class RecipeItem
    {
        public required int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public RecipeIngredientDto[]? Ingredients { get; set; }
        public RecipeStepDto[]? Steps { get; set; }
        public int? PrepTimeMinutes { get; set; }
        public int? CookTimeMinutes { get; set; }
        public int? Servings { get; set; }
        public required DateTime CreatedAt { get; set; }
        public DateTime? LastUpdatedAt { get; set; }
        public required int TimesCooked { get; set; }
        public DateTime? LastCooked { get; set; }
    }
}

File: orbit.app/interfaces/api/meal-planner/GetRecipesResponse.ts
import type { RecipeIngredient, RecipeStep } from './AddRecipeRequest'

export interface GetRecipesResponse {
  recipes: RecipeItem[]
}

export interface RecipeItem {
  id: number
  name: string
  description: string
  ingredients: RecipeIngredient[] | null
  steps: RecipeStep[] | null
  prepTimeMinutes: number | null
  cookTimeMinutes: number | null
  servings: number | null
  createdAt: string
  lastUpdatedAt: string | null
  timesCooked: number
  lastCooked: string | null
}
